backend/src/main/java/com/javahelp/backend/domain/user/authentication/TokenAuthResult.java

```diff
@@ -0,0 +1,45 @@
+package com.javahelp.backend.domain.user.authentication;
+
+import com.javahelp.model.token.Token;
+import com.javahelp.model.user.User;
+
+import org.graalvm.compiler.api.replacements.Snippet;
+
+public class TokenAuthResult {
+    private final String desiredUserID;
+    private final Token token;
+    private final boolean authenticated;
+    private final String errorMessage;
+
+    /**
+     * Constructs a TokenAuthResult object.
+     *
+     * @param desiredUserID: the ID of the user that needs to be authenticated
+     * @param token: the token used to authenticate the user
+     */
+    public TokenAuthResult(String desiredUserID, Token token) {
+        this.token = token;
+        this.desiredUserID = desiredUserID;
+        this.authenticated = desiredUserID.equals(token.getUserID());
+        if (authenticated) {
+            this.errorMessage = null;
+        } else {
+            this.errorMessage = "Token Authentication failed";
+        }
+    }
+
+    /**
+     * Constructor for when fetching the token from the database fails
+     *
+     * @param desiredUserID the ID of the user that needs to be authenticated
+     */
+    public TokenAuthResult(String desiredUserID) {
+        this.errorMessage = "Could not fetch from database";
+        this.desiredUserID = desiredUserID;
+        this.token = null;
+        this.authenticated = false;
+    }
+
+    public boolean getAuthenticated() {return this.authenticated;}
```

Comment:
Hmm Bill I think the Java doc for this method is still missing.

Reply:
Yeah he raised that question in the above thread

Reply:
Yeah I addressed it in the above thread, but I've just gone and added it since that would probably be the simplest solution

Reply:
Oh yeah my bad sorry Bill I didn't notice you replied to the previous post! Thanks for adding it! Yeah for ClientInfo I think I have Java doc for the setters and getters too but I don't have doc for the attributes.